addons/web_editor/static/src/js/editor/odoo-editor/test/spec/editor.test.js

```diff
@@ -4290,12 +4290,30 @@ X[]
         it('should transform url after space', async () => {
             await testEditor(BasicEditor, {
                 contentBefore: '<p>a http://test.com b http://test.com[] c http://test.com d</p>',
-                stepFunction: async (editor) => {
-                    editor.testMode = false;
-                    await insertText(editor, ' ');
-                },
-                contentAfter: '<p>a http://test.com b <a href="http://test.com">http://test.com</a>[] c http://test.com d</p>',
-                //in reality: '<p>a http://test.com b <a href="http://test.com">http://test.com</a>&nbsp;[] c http://test.com d</p>'
+                stepFunction: editor => insertText(editor, ' '),
+                contentAfter: '<p>a http://test.com b <a href="http://test.com">http://test.com</a> []&nbsp;c http://test.com d</p>',
+            });
+        });
+        it('should transform url followed by punctuation characters after space', async () => {
+            await testEditor(BasicEditor, {
+                contentBefore: '<p>http://test.com.[]</p>',
+                stepFunction: editor => insertText(editor, ' '),
+                contentAfter: '<p><a href="http://test.com">http://test.com</a>. []</p>',
+            });
+            await testEditor(BasicEditor, {
+                contentBefore: '<p>test.com...[]</p>',
+                stepFunction: editor => insertText(editor, ' '),
+                contentAfter: '<p><a href="http://test.com">test.com</a>... []</p>',
```

Comment:
Why are those URL suddently `http://` and not `https://` ?
It was not the case in #147174

Reply:
The default protocol for "normal" links was changed from `https` to `http` in 17.0 by https://github.com/odoo/odoo/commit/82ba5791f042498ba7e69a56cc8968d2b5912baf.

Reply:
See https://github.com/odoo/odoo/pull/147845/files#r1450315576